app/models/lti_configuration/blackboard_settings.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class LtiConfiguration
+  class BlackboardSettings < GenericSettings
+    def initialize
```

Comment:
There's another way of designing this that _I_ believe is a bit cleaner,

If you define each of `platform_name`, `icon`, etc as methods instead of `attr_readers` in `GenericSettings`, we can override those methods in these subclasses with our LMS-specific values.

It'd look something like this

```
class GenericSettings
  def icon
    # What is the default icon? Or leave empty if nil
  end
end

class BlackboardSettings < GenericSettings
  def icon
    "blackboard-logo.png"
  end
end
```

Each settings class overrides the specific methods it needs, and anything  not overridden falls back on the generic class. Defaults are defined in methods instead of a large hash or part of opts[:this] || "default" clauses. 

Feel free to ignore, I'm not sure what the preferred way of doing this is :)

Reply:
Good idea, I like that!